hugr-core/src/std_extensions/logic.rs

```diff
@@ -47,6 +47,13 @@ impl ConstFold for NaryLogic {
                 (res || inps.len() as u64 == num_args)
                     .then_some(vec![(0.into(), ops::Value::from_bool(res))])
             }
+            Self::Eq => {
+                let inps = read_inputs(consts)?;
+                let res = inps.clone().into_iter().reduce(|a, b| a == b)?;
```

Comment:
```suggestion
                let res = inps.iter().copied().reduce(|a, b| a == b)?;
```
Annoying you can't use `bool::eq` but indeed you can't; however `bool` impls `Copy`.

Reply:
(a) good to use Copy rather than Clone when available
(b) `clone().into_iter()` probably allocates a clone of the entire Vec, `iter().copied()` doesn't